### ISSUE TITLE
APT Module interpreted ssh output as failure

### ISSUE THREAD

**Issue by ghost** [ansible/ansible-modules-core#5090]
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

apt - package manager
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
roles_path = ../../../ansible/roles
retry_files_enabled = False
pipelining = True
timeout = 60
host_key_checking = False
forks = 10

[ssh_connection]
ssh_args = -o ControlMaster=no -o ServerAliveInterval=30 -F ssh_config
```
##### OS / ENVIRONMENT

Host:

```
Ubuntu 16.04
```

Guest:

```
Ubuntu 14.04
```
##### SUMMARY

I already created an issue in the main ansible github project, but I think this is a problem with the module. I tried already so many things to fix this problem, nothing helps so far. 

We are using Ansible to provision our openstack infrastructure. There are seperate playbooks and roles for provisioning the systems. One for basic software installation and configuration and one for maintaing the software configuration with the correct parameters. After a fresh build of an openstack stack, the provisioning playbooks is failing with an undefined "MODULE FAILURE" without stdout or stderr. These happens for example on installation nginx, as simple task.

```
- name: "install nginx"
  become: yes
  apt: name="nginx" state="present"
  notify: "restart nginx services"
```

The software was installed, but the build failed nevertheless. As I started the playbook again, there is no problem with this task but on another task the next package installation failed. When I activate the debug output (-vvvv) I can see a error:

```
[...]
status: The following connections are open:\r\n  #0 client-session (t4 r0 i3/0 o3/0 fd -1/-1 cc -1)\r\n\r\ndebug3: send packet: type 1\r\ndebug1: fd 1 clearing O_NONBLOCK\r\ndebug1: fd 2 clearing O_NONBLOCK\r\nConnection to 185.56.135.9 closed.\r\nTransferred: sent 3060, received 2604 bytes, in 15.8 seconds\r\nBytes per second: sent 194.0, received 165.1\r\ndebug1: Exit status 0\r\ndebug1: compress outgoing: raw data 806, compressed 512, factor 0.64\r\ndebug1: compress incoming: raw data 512, compressed 806, factor 1.57\r\n", "module_stdout": "\r\n", "msg": "MODULE FAILURE", "parsed": false
[...]
```

For me this looks like that ansible interpreted this as failure because he returns not a valid json string. You have any ideas to prevent this problem? Or have a fix?
##### STEPS TO REPRODUCE

It's not possible to reproduce these problem your own way, because we have a big collection of playbooks and roles.
##### EXPECTED RESULTS

```
changed: [dev] or ok: [dev]
```
##### ACTUAL RESULTS

```
"module_stdout": "\r\n", "msg": "MODULE FAILURE", "parsed": false
```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ghost** [comment]
After moving pipelining = True to the [ssh_connection] section, there aren't errors anymore.
